registry/purview-registry/script/migrate_legacy_project.py

```diff
@@ -0,0 +1,122 @@
+
+from typing import Union
+from uuid import UUID
+from azure.identity import DefaultAzureCredential
+from pyapacheatlas.auth.azcredential import AzCredentialWrapper
+from pyapacheatlas.core import (AtlasEntity, AtlasProcess,AtlasException,
+                                PurviewClient)
+from pyapacheatlas.core.util import GuidTracker
+
+Label_Contains = "CONTAINS"
+Label_BelongsTo = "BELONGSTO"
+Label_Consumes = "CONSUMES"
+Label_Produces = "PRODUCES"
+guid = GuidTracker(starting=-1000)
+
+def _generate_relation_pairs(from_entity, to_entity, relation_type):
+    global guid
+    type_lookup = {Label_Contains: Label_BelongsTo, Label_Consumes: Label_Produces}
+    forward_relation =  AtlasProcess(
+        name=str(from_entity["guid"]) + " to " + str(to_entity["guid"]),
+        typeName="Process",
+        qualified_name='__'.join(
+            [relation_type,str(from_entity["guid"]), str(to_entity["guid"])]),
+        inputs=[from_entity],
+        outputs=[to_entity],
+        guid=guid.get_guid())
+    
+    backward_relation = AtlasProcess(
+        name=str(to_entity["guid"]) + " to " + str(from_entity["guid"]),
+        typeName="Process",
+        qualified_name='__'.join(
+            [type_lookup[relation_type], str(to_entity["guid"]), str(from_entity["guid"])]),
+        inputs=[to_entity],
+        outputs=[from_entity],
+        guid=guid.get_guid())
+    return [forward_relation,backward_relation]
+def upload_single_entity_to_purview(purview_client,entity:Union[AtlasEntity,AtlasProcess]):
+    try:
+        entity.lastModifiedTS="0"
+        result = purview_client.upload_entities([entity])
+        entity.guid = result['guidAssignments'][entity.guid]
+        print(f"Successfully created {entity.typeName} -- {entity.qualifiedName}")
+    except AtlasException as e:
+        if "PreConditionCheckFailed" in e.args[0]:
+            entity.guid = purview_client.get_entity(qualifiedName=entity.qualifiedName,typeName = entity.typeName)['entities'][0]['guid']
+            print(f"Found existing entity  {entity.guid}, {entity.typeName} -- {entity.qualifiedName}")
+    return UUID(entity.guid)
+default_purview_name = "feathrazuretest3-purview1"   
```

Comment:
Is this hard coded? should we make it configurable?

Reply:
I will have some description to show the user how to run the script. Changing this string will be the first step